grails-app/controllers/au/org/emii/netcdfsubset/SubsetController.groovy

```diff
@@ -21,13 +23,29 @@ class SubsetController {
         response.setContentType("application/octet-stream")
         response.setHeader("Content-disposition", "filename=${filenameToServe(subsetRequest)}")
 
-        try {
-            subsetService.subset(subsetRequest.typeName, subsetRequest.CQL_FILTER, response.outputStream)
+        if (isThreadAvailable()) {
```

Comment:
(I wrote the last comment independently of what @julian1 wrote - but we saw the same thing).